src/main/scala/com/lightbend/kafka/scala/streams/StreamsBuilderS.scala

```diff
@@ -17,62 +17,65 @@ import scala.collection.JavaConverters._
 
-  def table[K, V](topic: String) : KTableS[K, V] = inner.table[K, V](topic)
+  def table[K, V](topic: String) : KTableS[K, V] = streamsBuilder.table[K, V](topic)
 
   def table[K, V](topic: String, consumed: Consumed[K, V]) : KTableS[K, V] =
-    inner.table[K, V](topic, consumed)
+    streamsBuilder.table[K, V](topic, consumed)
 
   def table[K, V](topic: String, consumed: Consumed[K, V],
-    materialized: Materialized[K, V, KeyValueStore[Bytes, Array[Byte]]]): KTableS[K, V] = 
-    inner.table[K, V](topic,  consumed, materialized)
+    materialized: Materialized[K, V, KeyValueStore[Bytes, Array[Byte]]]): KTableS[K, V] =
+    streamsBuilder.table[K, V](topic,  consumed, materialized)
 
-  def table[K, V](topic: String, 
-    materialized: Materialized[K, V, KeyValueStore[Bytes, Array[Byte]]]): KTableS[K, V] = 
-    inner.table[K, V](topic, materialized)
+  def table[K, V](topic: String,
+    materialized: Materialized[K, V, KeyValueStore[Bytes, Array[Byte]]]): KTableS[K, V] =
+    streamsBuilder.table[K, V](topic, materialized)
 
   def globalTable[K, V](topic: String): GlobalKTable[K, V] =
-    inner.globalTable(topic)
+    streamsBuilder.globalTable(topic)
 
-  def globalTable[K, V](topic: String, consumed: Consumed[K, V]) : GlobalKTable[K, V] = 
-     inner.globalTable(topic, consumed)
+  def globalTable[K, V](topic: String, consumed: Consumed[K, V]) : GlobalKTable[K, V] =
+     streamsBuilder.globalTable(topic, consumed)
 
   def globalTable[K, V](topic: String, consumed: Consumed[K, V],
-    materialized: Materialized[K, V, KeyValueStore[Bytes, Array[Byte]]]): GlobalKTable[K, V] = 
-    inner.globalTable(topic,  consumed, materialized)
+    materialized: Materialized[K, V, KeyValueStore[Bytes, Array[Byte]]]): GlobalKTable[K, V] =
+    streamsBuilder.globalTable(topic,  consumed, materialized)
 
-  def globalTable[K, V](topic: String, 
-    materialized: Materialized[K, V, KeyValueStore[Bytes, Array[Byte]]]): GlobalKTable[K, V] = 
-    inner.globalTable(topic,  materialized)
+  def globalTable[K, V](topic: String,
+    materialized: Materialized[K, V, KeyValueStore[Bytes, Array[Byte]]]): GlobalKTable[K, V] =
+    streamsBuilder.globalTable(topic,  materialized)
 
-  def addStateStore(builder: StoreBuilder[_ <: StateStore]): StreamsBuilder = inner.addStateStore(builder)
+  def addStateStore(builder: StoreBuilder[_ <: StateStore]): StreamsBuilder = streamsBuilder.addStateStore(builder)
 
   def addGlobalStore(storeBuilder: StoreBuilder[_ <: StateStore], topic: String, sourceName: String, consumed: Consumed[_, _], processorName: String, stateUpdateSupplier: ProcessorSupplier[_, _]): StreamsBuilder =
-    inner.addGlobalStore(storeBuilder,topic,sourceName,consumed,processorName,stateUpdateSupplier)
+    streamsBuilder.addGlobalStore(storeBuilder,topic,sourceName,consumed,processorName,stateUpdateSupplier)
 
-  def build() : Topology = inner.build()
+  def build() : Topology = streamsBuilder.build()
 }
 
+object StreamsBuilderS {
+
```

Comment:
On concern is that the existing API will break. Those who are using `new StreamsBuilderS()` will need to change to the object syntax. But I guess this will be a worthwhile change and we are only in `0.1.0`. Thoughts @deanwampler, @lightbend/fast-data ?

Reply:
I actually like the previous version more. In the proposed PR we kinda mixing Java and Scala APIs. Any specific reason for this?

Reply:
The problem with the earlier version is that the API cannot be used if someone wants to use the Scala API for a mocked `StreamsBuilder`, say, for unit testing. Or someone may have a specialized version of `StreamsBuilder` derived from the Java base `StreamsBuilder`. She cannot use it with the Scala API since we are constructing the base Java `StreamsBuilder` within the class itself. S, it's always recommended to pass in the instance (constructor injection) instead of creating the hardcoded instance inside.